partiql-tests-kotlin-test-runner/src/test/kotlin/org/partiql/tests/kotlintestrunner/TestRunner.kt

```diff
@@ -0,0 +1,283 @@
+/*
+ * Copyright 2022 Amazon.com, Inc. or its affiliates.  All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at:
+ *
+ *       http://aws.amazon.com/apache2.0/
+ *
+ *  or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific
+ *  language governing permissions and limitations under the License.
+ */
+
+package org.partiql.tests.kotlintestrunner
+
+import com.amazon.ion.system.IonSystemBuilder
+import org.junit.jupiter.params.ParameterizedTest
+import org.junit.jupiter.params.provider.ArgumentsSource
+import org.partiql.lang.CompilerPipeline
+import org.partiql.lang.SqlException
+import org.partiql.lang.eval.CompileOptions
+import org.partiql.lang.eval.EvaluationSession
+import org.partiql.lang.eval.TypingMode
+import java.io.File
+
+private val PARTIQL_EVAL_TEST_DATA_DIR = System.getenv("PARTIQL_EVAL_TESTS_DATA")
+private val PARTIQL_EVAL_EQUIV_TEST_DATA_DIR = System.getenv("PARTIQL_EVAL_EQUIV_TESTS_DATA")
+
+private val ION = IonSystemBuilder.standard().build()
+
+private val COERCE_EVAL_MODE_COMPILE_OPTIONS = CompileOptions.build { typingMode(TypingMode.PERMISSIVE) }
+private val ERROR_EVAL_MODE_COMPILE_OPTIONS = CompileOptions.build { typingMode(TypingMode.LEGACY) }
+
+private val LANG_KOTLIN_EVAL_SKIP_LIST = listOf(
```

Comment:
The skip lists defined in this file show how the current Kotlin implementation diverges from the PartiQL spec. Most of the divergent behavior is due to `partiql-lang-kotlin` not having a STRICT/ERROR typing mode.  The [LEGACY typing mode](https://github.com/partiql/partiql-lang-kotlin/blob/main/lang/src/org/partiql/lang/eval/CompileOptions.kt#L53-L62) (which is closer to STRICT/ERROR but not a complete match) was used for testing the STRICT/ERROR typing mode behavior.

A lot of the other behavior differences is due to not supporting some syntax mentioned in the spec (like `COLL_*` aggregation functions) and due to not supporting coercions.

The remaining divergent behavior causing certain conformance tests to fail are likely bugs, which I plan to cut an issue to track once we confirm the conformance test behavior is correct.

Reply:
Honestly, the entire explanation here might be useful as a comment in this file.

Reply:
Agree that a comment is necessary here. Added in new commit.